rskj-core/src/main/java/co/rsk/config/RemascConfigFactory.java

```diff
@@ -34,24 +42,57 @@ public class RemascConfigFactory {
     private static final Logger logger = LoggerFactory.getLogger("RemascConfigFactory");
 
     private ObjectMapper mapper;
-    private String configPath;
+    private final Map<String, RemascConfig> remascConfigMap = new ConcurrentHashMap<>();
 
     public RemascConfigFactory(String remascConfigFile) {
         this.mapper = new ObjectMapper();
-        this.configPath = remascConfigFile;
+        logger.info("RemascConfigFactory remascConfigFile = {} ", remascConfigFile);
+
+        try (InputStream is = RemascConfigFactory.class.getClassLoader().getResourceAsStream(remascConfigFile)) {
+            String sis = new BufferedReader(new InputStreamReader(is, StandardCharsets.UTF_8))
+                    .lines()
+                    .collect(Collectors.joining("\n"));
+
+            logger.info("RemascConfigFactory sis = {} ", sis);
+
+            JsonNode configNode = JacksonParserUtil.readTree(mapper, sis);
+
+            logger.info("RemascConfigFactory node = {} ", configNode);
+
+            Iterator<String> fieldNames = configNode.fieldNames();
+
+            logger.info("RemascConfigFactory fieldNames = {} ", fieldNames);
+
+            fieldNames.forEachRemaining(field -> remascConfigMap.put(field, this.fetchRemascConfigMap(field, configNode)));
+
+            logger.info("RemascConfigFactory remascConfigMap = {} ", remascConfigMap);
+        } catch (Exception ex) {
+            logger.error("Error reading REMASC", ex);
+            throw new RemascException("Error reading REMASC [" + remascConfigFile + "]: ", ex);
+        }
     }
 
-    public RemascConfig createRemascConfig(String config) {
+    private RemascConfig fetchRemascConfigMap(String config, JsonNode configNode) {
         RemascConfig remascConfig;
 
-        try (InputStream is = RemascConfigFactory.class.getClassLoader().getResourceAsStream(this.configPath)) {
-            JsonNode node = JacksonParserUtil.readTree(mapper, is);
-            remascConfig = JacksonParserUtil.treeToValue(mapper, node.get(config), RemascConfig.class);
+        logger.info("fetchRemascConfigMap config = {} ", config);
+
+        try {
+            logger.info("fetchRemascConfigMap node.get(config) = {} ", configNode.get(config));
+
+            remascConfig = JacksonParserUtil.treeToValue(mapper, configNode.get(config), RemascConfig.class);
+
+            logger.info("fetchRemascConfigMap remascConfig = {} ", remascConfig);
         } catch (Exception ex) {
```

Comment:
do we really need to catch this exception in this private method? It's being done at the upper level too.

Reply:
Addessed